src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -3231,11 +3222,58 @@ void StorageReplicatedMergeTree::cloneReplica(const String & source_replica, Coo
                 created_get_parts.insert(part_name);
         }
 
-        LOG_TEST(log, "Copying entry {}", entry_info.data);
-        zookeeper->create(fs::path(replica_path) / "queue/queue-", entry_info.data, zkutil::CreateMode::PersistentSequential);
+        copy_entries.push_back(i);
         ++total_entries_to_copy;
     }
 
+    std::vector<zkutil::ZooKeeper::FutureCreate> futures;
+    futures.reserve(mimic_entries.size() + copy_entries.size());
+    auto zookeeper_ops_timeout = std::chrono::milliseconds(zookeeper->getOperationTimeoutMS());
+    for (auto & [entry, future] : mimic_entries)
+    {
+        if (future.valid())
+        {
+            /// Not throwing exception because entry->create_time is optional field
+            if (future.wait_for(zookeeper_ops_timeout) == std::future_status::ready)
+            {
+                auto response = future.get();
+                Coordination::Error code = response.error;
+                if (code == Coordination::Error::ZOK)
+                    entry->create_time = response.stat.ctime / 1000;
+            }
+            else
+            {
+                zookeeper->finalize(fmt::format("Operation timeout on {} {}", Coordination::OpNum::Exists, fs::path(replica_path) / "parts" / entry->new_part_name));
+            }
+        }
+        LOG_TEST(log, "Enqueueing {} for fetch", entry->new_part_name);
+        futures.emplace_back(zookeeper->asyncCreate(fs::path(replica_path) / "queue/queue-", entry->toString(), zkutil::CreateMode::PersistentSequential));
+    }
+
+    for (const auto & index : copy_entries)
+    {
+        LOG_TEST(log, "Copying entry {}", source_queue[index].data);
+        futures.emplace_back(zookeeper->asyncCreate(fs::path(replica_path) / "queue/queue-", source_queue[index].data, zkutil::CreateMode::PersistentSequential));
```

Comment:
Consider batching create requests and sending them as a multi request (however, Apache ZooKeeper has 1MB limit for multi-request size, but sending MULTI_BATCH_SIZE requests at once should be fine)